src/server.ts

```diff
@@ -63,6 +63,32 @@ const build = (opts: FastifyServerOptions): FastifyInstance => {
       return;
     }
 
+    const authHeader = req.headers.authorization as string;
+    if (authHeader && authHeader.startsWith('Basic ')) {
+      if (!req.routeOptions.url?.startsWith('/api')) {
```

Comment:
When i POST the config without credentials it will try to redirect me to the `/login` endpoint which i think is awkward for an API. Is there any way this can be improved?

Reply:
resolved